src/index.js

```diff
@@ -360,19 +359,14 @@ const compile = (schema, root, opts, scope, basePathRoot) => {
 
     /* Preparation and methods, post-$ref validation will begin at the end of the function */
 
-    const { type } = node
-    if (!type) enforceValidation('type is required')
-    if (type !== undefined && typeof type !== 'string' && !Array.isArray(type))
-      fail('Unexpected type')
```

Comment:
This check is missing now - is it no longer necessary?

Reply:
It's already checked in
```js
     for (const t of typeArray || [])
       enforce(typeof t === 'string' && types.has(t), 'Unknown type:', t)
```